### PR TITLE
Updates Units::isAnimal

### PR DESCRIPTION
**Pull request by cppcooper**
(edit)
Now checks that the unit cannot learn or speak, and has either a wilderness population source, a pet/pet_exotic token, or is trainable for war/hunting

This mostly matches the Dwarf Therapist animal checks. Only we care about wild animals too.

### CODE REVIEW

**Review by myk002 — COMMENTED** [review]
changelog plz

**Comment by cppcooper** [comment]
> changelog plz

The isAnimal call isn't in a release yet, the existing changelog for it isn't sufficient?

File: library/modules/Units.cpp [hunk 1]
```diff
@@ -626,7 +626,8 @@ bool Units::isDwarf(df::unit *unit)
 bool Units::isAnimal(df::unit* unit)
 {
     CHECK_NULL_POINTER(unit)
-    return unit->enemy.caste_flags.is_set(df::enums::caste_raw_flags::NATURAL_ANIMAL);
+    return unit->animal.leave_countdown != 0 || 
+           unit->enemy.caste_flags.is_set(df::enums::caste_raw_flags::NATURAL_ANIMAL);
```
Comment (ab9rf) [inline library/modules/Units.cpp:630]: I am not convinced that `NATURAL_ANIMAL` is an appropriate check condition. Per the DF wiki, `NATURAL_ANIMAL` is an alias of `NATURAL` and `NATURAL` means:

> Animal is considered to be natural. NATURAL animals will not engage creatures tagged with `AT_PEACE_WITH_WILDLIFE` in combat unless they are members of a hostile entity and vice-versa.

This doesn't seem relevant to the purpose of this predicate.

**Review by ab9rf — APPROVED** [review]
we've discussed this extensively on discord. the test presented draws from my own reverse-engineering of the unitlist viewscreen, experimental evidence collected by cppcooper and rumrusher, and code in dwarf therapist. i'm satisfied that it's reasonable.
